Assets/Scripts/EnemyController.cs

```diff
@@ -43,4 +42,15 @@ private void FixedUpdate()
         }
         Movement(targetPos, Vector2.zero);
     }
+
+    protected void OnCollisionEnter2D(Collision2D collision)
```

Comment:
I don't believe OnCollision to be great here, considering it isn't consistent when units are "hugging" each other.
What I'd suggest is to create a trigger collider for a hitbox, and then use OnTrigger methods instead. Bonus points if you can make either the hitbox point at the target, or simply deal damage only to the target, regardless of what other entities are inside the trigger. Screenshot example of a trigger hitbox (notice the isTrigger variable in the inspector)
![image](https://github.com/hsven/mystic-militia/assets/72400318/c991eaf2-71a0-4fac-84b9-531c7111abe6)

Reply:
That's the prefab's own "view". Just double click on the prefab object

Reply:
I worked on the hitbox, it's working now, what do you think about it ?

Reply:
The trigger hitbox feels better, but notice how the opposing units still don't do anything when "hugging" each other.
This is because you used `OnTriggerEnter2D`, which only procs once, when a collider enters another. This means that if they hug, then don't move much anymore, they only dealt damage to each other once.

Check out [OnTriggerStay2D](https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnTriggerStay2D.html). By itself it won't be good cuz it procs every frame, but if you add a cooldown or something, it might fix this.

Reply:
Ok, I will have a look on that.